src/utils/responsiveProp.ts

```diff
@@ -0,0 +1,47 @@
+const breakpointToMinPxMapping = {
+  small: 0,
+  medium: 768,
+  large: 1024,
+} satisfies Record<string, number>
+
+type Breakpoint = keyof typeof breakpointToMinPxMapping
+const breakpointList = Object.keys(breakpointToMinPxMapping) as Breakpoint[]
+
+type PropByBreakpoint<PropValue> = { [key in Breakpoint]?: PropValue }
+export type ResponsiveProp<PropValue> = PropValue | PropByBreakpoint<PropValue>
+
+const isPropByBreakpoint = <PropValue>(
+  value: ResponsiveProp<PropValue>,
+): value is PropByBreakpoint<PropValue> => {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    breakpointList.includes(Object.keys(value)[0] as any)
+  )
+}
+
+export const resolveResponsiveProp = <PropValue>(
+  value: ResponsiveProp<PropValue>,
+  styleTemplate: (value: PropValue) => string,
+) => {
+  if (!isPropByBreakpoint(value)) {
```

Comment:
oh so this is what you were mentioning around having to set it in a media query anyway just to keep the correct order of the styles.

Reply:
Yeah, you can try to remove the media query, it will change the behaviour

Reply:
So I guess we just need to be sure if we wanna add more breakdpoints to have the lowest one always at 0 for this edge case